library.py

```diff
@@ -204,5 +204,5 @@ def src_dir(*x):
     SRC_FILTER=sources
 )
 
-global_env.Replace(
-    LDSCRIPT_PATH=linker_script)
+env.Append(
+    LINKFLAGS=['-T',linker_script])
```

Comment:
So, now that the custom board issue is fixed, can this be restored to `global_env.Replace`? That way we shouldn't have the double `-T` flag issue.

Reply:
Ok, I ran into the same problem with the 4.1.0 upgrade. How about we remove the -T flag? I fixed it using the following line to remove the -T linker_script.ld

```
global_env['LINKFLAGS'] = [ x for x in global_env['LINKFLAGS'] if not ('-T' == x or ".ld" in x)]
```
Then the replace works. (This isn't the best remove code - expects linker file to have .ld extension.)

Reply:
looks good to me! indeed it's best to check if there is already a -T flag in the link flags and if so, replace it with the new linker script.